mathics/builtin/files_io/files.py

```diff
@@ -20,6 +20,7 @@
 
 from mathics_scanner import TranslateError
 from mathics.core.parser import MathicsFileLineFeeder, parse
+from mathics.core import read
```

Comment:
Usually, we don't import modules, but just the things we need inside those modules, doing this makes circular imports less common.

```py
from mathics.core.read import INPUTFILE_VAR
```

Reply:
I tried this, but it does not work for me on Python 3.8.10.
According to [this discussion](https://stackoverflow.com/questions/13034496/using-global-variables-between-files#comment104154371_45646477) I am not the only one observing this behavior.
Furthermore this is also the style used for the global variables in `settings.py`.

Reply:
While whether or not `from x import` or `import` is used (and in my view this kind of thing is debatable), in Python this has no effect on circular imports. 

I tried a little test and in that it also shows why this doesn't change things: 

File a.py: 
```
from b import B
A = "A" + B
```

File b.py:

```
from a import A
b = "B" + A
```

File `__main__.py`: 

```
from a import A
from b import B

print(A + B)
```

The usual null `__init__.py` (but I am not sure that is needed here). 

Running `__main__.py`: 

```
python __main__.py 
Traceback (most recent call last):
  File "__main__.py", line 1, in <module>
    from a import A
  File "/tmp/myfoo/a.py", line 1, in <module>
    from b import B
  File "/tmp/myfoo/b.py", line 1, in <module>
    from a import A
ImportError: cannot import name 'A' from partially initialized module 'a' (most likely due to a circular import) (/tmp/myfoo/a.py)

```

Reply:
@hrueter ok, thanks for explaining.

Reply:
The reason to do not import full modules but just the needed elements is that it helps to debug these circular references, not because it avoids it.